### PR TITLE
test: Use tempdir to produce BB vis output

### PR DESCRIPTION
**Pull request by paulgessinger**
Use a temporary directory to store visualization outputs from the BoundingBox test.

### CODE REVIEW

File: Tests/UnitTests/Core/Utilities/BoundingBoxTest.cpp [hunk 1]
```diff
@@ -279,7 +293,7 @@ BOOST_AUTO_TEST_CASE(intersect_rays) {
     PlyVisualization3D<BoundingBoxScalar> ply;
 
     ray3.draw(ply);
-    std::ofstream os("ray3d.ply");
+    std::ofstream os(tmp("ray3d.ply"));
```
Comment (andiwand) [inline Tests/UnitTests/Core/Utilities/BoundingBoxTest.cpp:296]:
```suggestion
    auto os = tmp("ray3d.ply");
```

would look more natural to me but feel free to ignore

**Review by andiwand — APPROVED** [review]

File: Tests/UnitTests/Core/Utilities/BoundingBoxTest.cpp [hunk 2]
```diff
@@ -442,7 +456,7 @@ BOOST_AUTO_TEST_CASE(ray_obb_intersect) {
 BOOST_AUTO_TEST_CASE(frustum_intersect) {
   BOOST_TEST_CONTEXT("2D") {
     auto make_svg = [](std::string fname, size_t w, size_t h) {
-      std::ofstream os(fname);
+      std::ofstream os(tmp(fname));
```
Comment (paulgessinger) [inline Tests/UnitTests/Core/Utilities/BoundingBoxTest.cpp:459]:
```suggestion
      auto os = tmp(fname);
```

File: Tests/UnitTests/Core/Utilities/BoundingBoxTest.cpp [hunk 3]
```diff
@@ -653,7 +667,7 @@ BOOST_AUTO_TEST_CASE(frustum_intersect) {
     //}
     // os.close();
 
-    // os = std::ofstream("frust3D_angle.ply");
+    // os = std::ofstream(tmp("frust3D_angle.ply"));
```
Comment (paulgessinger) [inline Tests/UnitTests/Core/Utilities/BoundingBoxTest.cpp:670]:
```suggestion
    // os = tmp("frust3D_angle.ply");
```

File: Tests/UnitTests/Core/Utilities/BoundingBoxTest.cpp [hunk 4]
```diff
@@ -843,7 +857,7 @@ BOOST_AUTO_TEST_CASE(frustum_intersect) {
       std::stringstream ss;
       ss << "frust3d-3s_test_" << l << ".ply";
 
-      os = std::ofstream(ss.str());
+      os = std::ofstream(tmp(ss.str()));
```
Comment (paulgessinger) [inline Tests/UnitTests/Core/Utilities/BoundingBoxTest.cpp:860]:
```suggestion
      os = tmp(ss.str());
```

File: Tests/UnitTests/Core/Utilities/BoundingBoxTest.cpp [hunk 5]
```diff
@@ -902,7 +916,7 @@ BOOST_AUTO_TEST_CASE(frustum_intersect) {
     // size_t n_vtx    = 1;
 
     // helper.clear();
-    // os = std::ofstream("frust3d-4s_dir.ply");
+    // os = std::ofstream(tmp("frust3d-4s_dir.ply"));
```
Comment (paulgessinger) [inline Tests/UnitTests/Core/Utilities/BoundingBoxTest.cpp:919]:
```suggestion
    // os = tmp("frust3d-4s_dir.ply");
```

File: Tests/UnitTests/Core/Utilities/BoundingBoxTest.cpp [hunk 6]
```diff
@@ -928,7 +942,7 @@ BOOST_AUTO_TEST_CASE(frustum_intersect) {
 
     // os << helper << std::flush;
     // os.close();
-    // os = std::ofstream("frust3d-4s_angle.ply");
+    // os = std::ofstream(tmp("frust3d-4s_angle.ply"));
```
Comment (paulgessinger) [inline Tests/UnitTests/Core/Utilities/BoundingBoxTest.cpp:945]:
```suggestion
    // os = tmp("frust3d-4s_angle.ply");
```

File: Tests/UnitTests/Core/Utilities/BoundingBoxTest.cpp [hunk 7]
```diff
@@ -1107,7 +1121,7 @@ BOOST_AUTO_TEST_CASE(frustum_intersect) {
       std::stringstream ss;
       ss << "frust3d-4s_test_" << l << ".ply";
 
-      os = std::ofstream(ss.str());
+      os = std::ofstream(tmp(ss.str()));
```
Comment (paulgessinger) [inline Tests/UnitTests/Core/Utilities/BoundingBoxTest.cpp:1124]:
```suggestion
      os = tmp(ss.str());
```

File: Tests/UnitTests/Core/Utilities/BoundingBoxTest.cpp [hunk 8]
```diff
@@ -1169,7 +1183,7 @@ BOOST_AUTO_TEST_CASE(frustum_intersect) {
 
     BOOST_CHECK(bb.intersect(fr));
 
-    std::ofstream os("frust3d-5s.ply");
+    std::ofstream os(tmp("frust3d-5s.ply"));
```
Comment (paulgessinger) [inline Tests/UnitTests/Core/Utilities/BoundingBoxTest.cpp:1186]:
```suggestion
    auto os = tmp("frust3d-5s.ply");
```

File: Tests/UnitTests/Core/Utilities/BoundingBoxTest.cpp [hunk 9]
```diff
@@ -1195,7 +1209,7 @@ BOOST_AUTO_TEST_CASE(frustum_intersect) {
 
     BOOST_CHECK(bb.intersect(fr));
 
-    std::ofstream os("frust3d-10s.ply");
+    std::ofstream os(tmp("frust3d-10s.ply"));
```
Comment (paulgessinger) [inline Tests/UnitTests/Core/Utilities/BoundingBoxTest.cpp:1212]:
```suggestion
    auto os = tmp("frust3d-10s.ply");
```

File: Tests/UnitTests/Core/Utilities/BoundingBoxTest.cpp [hunk 10]
```diff
@@ -1220,7 +1234,7 @@ BOOST_AUTO_TEST_CASE(frustum_intersect) {
 
     BOOST_CHECK(bb.intersect(fr));
 
-    std::ofstream os("frust3d-4s-bigbox.ply");
+    std::ofstream os(tmp("frust3d-4s-bigbox.ply"));
```
Comment (paulgessinger) [inline Tests/UnitTests/Core/Utilities/BoundingBoxTest.cpp:1237]:
```suggestion
    auto os = tmp("frust3d-4s-bigbox.ply");
```

**Review by andiwand — APPROVED** [review]

**Pull request merged**